junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java

```diff
@@ -1576,11 +1576,42 @@ public static void assertIterableEquals(Iterable<?> expected, Iterable<?> actual
 	 * >> M A N Y  M O R E  E N T R I E S >>
 	 * drwxr-xr-x  0 root root   512 Sep 22  2017 var
 	 * }</pre>
+	 * <p>Fails with a generated failure message describing the difference.
 	 */
 	public static void assertLinesMatch(List<String> expectedLines, List<String> actualLines) {
 		AssertLinesMatch.assertLinesMatch(expectedLines, actualLines);
 	}
 
+	/**
+	 * <em>Assert</em> that {@code expected} list of {@linkplain String}s matches {@code actual}
+	 * list.
+	 *
+	 * <p>Find a detailed description of the matching algorithm here: {@linkplain #assertLinesMatch(List, List)}
+	 *
+	 * <p>Fails with the supplied failure {@code message} and the generated message.
+	 *
+	 * @see #assertLinesMatch(List, List)
+	 */
+	public static void assertLinesMatch(List<String> expectedLines, List<String> actualLines, String message) {
+		AssertLinesMatch.assertLinesMatch(expectedLines, actualLines, message);
+	}
+
+	/**
+	 * <em>Assert</em> that {@code expected} list of {@linkplain String}s matches {@code actual}
+	 * list.
+	 *
+	 * <p>Find a detailed description of the matching algorithm here: {@linkplain #assertLinesMatch(List, List)}
+	 *
+	 * <p>If necessary, a custom failure message will be retrieved lazily from the supplied
+	 * {@code messageSupplier}. Fails with the supplied and the generated failure message.
```

Comment:
```suggestion
	 * {@code messageSupplier}. Fails with the custom failure message prepended to
	 * a generated failure message describing the difference.
```

Reply:
Looks like I submitted my review before submitting the same suggestion for the other method. So please just copy-n-paste this change to the other method. 

TIA